### DISCUSSION TITLE
QUESTION｜由于编码错误导致无法读取文件

### DISCUSSION THREAD

**Discussion by LiuJiewenTT** [MCSLTeam/MCSL2#139] [提问区域]
### MCSL2 系统报告

Traceback (most recent call last):
  File "D:\Unzipped Software\mc\MCSL2\MCSL2-~1.1-W\MCSL2Lib\Pages\consolePage.py", line 355, in colorConsoleText
  File "D:\Unzipped Software\mc\MCSL2\MCSL2-~1.1-W\MCSL2Lib\Controllers\serverController.py", line 395, in readServerProperties
UnicodeDecodeError: 'gbk' codec can't decode byte 0x80 in position 656: illegal multibyte sequence

### 你在哪方面遇到了问题？

MCSL2启动服务器

### 说明

server.properties不能包含中文吗？我想设置中文的motd

### 附件(可选)

![image](https://github.com/MCSLTeam/MCSL2/assets/43092274/a37ceae3-a3b4-40e2-8f03-5029af0838d9)

### 确认

- [X] 我确保我的Issue没有和别人的重复

**Comment by lxhtt** [comment]
请使用UTF-8格式保存`server.properties`文件。

**Reply by LiuJiewenTT** [reply]
没用，已经用vscode保存为utf8了，还是报错gbk无法解码。这应该是open函数没有指定编码为utf8的问题。

**Reply by LiuJiewenTT** [reply]
即便用gbk保存，进游戏看到的motd内容还是乱码。
![image](https://github.com/MCSLTeam/MCSL2/assets/43092274/6bc512da-c210-4731-841b-2e961a1f9467)

**Reply by LiuJiewenTT** [reply]
即便保存成18030，开了服内容就变这样：
![image](https://github.com/MCSLTeam/MCSL2/assets/43092274/a9e65a4c-9880-4ba0-adc1-9ad01677e1e0)

保存成utf8根本无法启动！

**Reply by lxhtt** [reply]
了解了，请等待修复。

**Comment by lxhtt** [comment]
Fixed in https://github.com/MCSLTeam/MCSL2/commit/3afcf97a147460f60b384719f2d6c06001550651
